mesonbuild/dependencies/boost.py

```diff
@@ -652,7 +649,7 @@ def filter_libraries(self, libs: T.List[BoostLibraryFile], lib_vers: str) -> T.L
     def detect_libraries(self, libdir: Path) -> T.List[BoostLibraryFile]:
         libs = []  # type: T.List[BoostLibraryFile]
         for i in libdir.iterdir():
-            if not i.is_file() or i.is_symlink():
```

Comment:
Symlinks were excluded on purpose to avoid duplicate libraries in the detection logic and because symlinks usually remove information. Please either ensure that all symlinks are resolved or that "real" paths of a library are always preferred.

Reply:
I changed it to resolve all the symlinks and use set() to remove duplicates

Reply:
Why not make libs into a set from the very beginning?

Reply:
No good reason. I changed it to use a set from the begining